lib/cache/storages/MemoryStorage.ts

```diff
@@ -0,0 +1,32 @@
+import { Storage } from './Storage';
+
+export class MemoryStorage implements Storage {
+
+  private readonly storage = new Map<string, any>();
+
+  constructor(private readonly limit?: number) { }
+
+  public set<V>(key: string, value: V): Storage {
+    if (typeof this.limit === 'number' && this.storage.size >= this.limit) {
+      return this;
+    }
```

Comment:
I'd expect that when limit is exceeded, then the value is still cached, but the oldest entry is removed.  That means that when a limit is specified, you also use a bounded list to track which values must be discarded.